src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/AsyncCollectorBindingProvider.cs

```diff
@@ -207,7 +207,11 @@ public IEnumerable<BindingRule> GetRules()
         }
 
         // Listed in precedence for providing via DefaultType.
-        private static readonly Type[] _defaultTypes = new Type[] { typeof(JObject), typeof(byte[]), typeof(string) };
+        // Precdence is more important than how we produce the default type (a direct conversion vs. a converter)
+        private static readonly Type[] _defaultTypes = new Type[] {
```

Comment:
See corresponding logic in Script here: https://github.com/Azure/azure-webjobs-sdk-script/blob/3f54f80a447e9330cb11b6a4ff203bcfdd6ba0ab/src/WebJobs.Script/Binding/FunctionBinding.cs#L152